### ISSUE TITLE
Body cannot be parsed, because req.__proto__ is wrong

### ISSUE THREAD

**Issue by harnyk** [Zakjholt/micro-cookie#4]
Consider the following snippet

```js

const parseCookies = require('micro-cookie');
const { buffer } = require('micro');

module.exports = parseCookies(async (req, res) => {
    const rawBody = await buffer(req); // <--- throws error "stream.on is not a function"
});
```

Body cannot be parsed, because `req` is not an instance of `http.IncomingMessage` anymore after the handler was wrapped with `parseCookies`.

**Comment by philcluff** [comment]
Anyone have a workaround for this?

Edit: I realized this module was 9 lines, mostly boilerplate, so I'm just using https://github.com/jshttp/cookie directly instead. Hope this helps someone.

**Comment by hitochan777** [comment]
I also had a same issue. The problem lies in the line in `index.js` as shown below.

```javascript
const newReq = Object.assign({}, req, { cookies });
```
This line is supposed to copy request object and assign cookies field to it. But `Object.assign` can only be used to copy enumerable own properties (See [official doucument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)).
So instead we have to do like this:

```javascript
res.cookies = cookies
```
And actually this is how expressjs cookie-parser is [implemented](https://github.com/expressjs/cookie-parser/blob/master/index.js#L60).

I am using the following cookie handler and it seems to be working.

```javascript
const cookie = (handler: Handler) => (req, res, ...restArgs) => {
  const cookies = parse((req.headers && req.headers.cookie) || "")
  req.cookies = cookies
  return handler(req, res, ...restArgs);
};
```

**Comment by zakjholt** [comment]
Should be resolved in latest version 1.2
